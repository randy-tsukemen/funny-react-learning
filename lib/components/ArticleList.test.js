import React from "react";
import ArticleList from "./ArticleList";
import renderer from "react-test-renderer";
// import { shallow } from "enzyme";

describe("ArticleList", () => {
    const testProps = {
        articles: {
            a: { id: "a" },
            b: { id: "b" },
        },
        articleActions: {
            lookupAuthor: jest.fn(() => ({})),
        },
    };
    it("should render a list of articles", () => {
        const tree = renderer.create(<ArticleList {...testProps} />).toJSON();

        console.log(tree);
        expect(tree.children.length).toBe(2);
        expect(tree).toMatchSnapshot();
        // expect(wrapper.find("article")).to.have.length(3);
    });
});
