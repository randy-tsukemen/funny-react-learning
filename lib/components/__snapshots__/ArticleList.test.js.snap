// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should render a list of articles 1`] = `
<div>
  <div
    className="article"
  >
    <div />
    <div>
      Invalid Date
    </div>
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
  <div
    className="article"
  >
    <div />
    <div>
      Invalid Date
    </div>
    <div>
      <a>
         
      </a>
    </div>
    <div />
  </div>
</div>
`;
